src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenSubmitApplication.java

```diff
@@ -37,29 +39,32 @@
     private PaymentService paymentService;
 
     @Override
-    public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
 
         configBuilder
             .event(CITIZEN_SUBMIT)
-            .forStates(Draft)
+            .forStates(Draft, AwaitingPayment)
             .name("Applicant 1 Statement of Truth")
             .description("Applicant 1 confirms SOT")
-            .aboutToStartCallback(this::aboutToStart)
             .retries(120, 120)
-            .grant(CREATE_READ_UPDATE, CITIZEN)
-            .grant(READ, CASEWORKER_DIVORCE_SUPERUSER);
+            .grant(CREATE_READ_UPDATE, CITIZEN, CASEWORKER_DIVORCE_COURTADMIN, CASEWORKER_DIVORCE_COURTADMIN_BETA)
+            .grant(READ, CASEWORKER_DIVORCE_SUPERUSER)
+            .aboutToSubmitCallback(this::aboutToSubmit);
     }
 
-    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDetails<CaseData, State> details) {
-
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
         log.info("Submit application about to start callback invoked");
 
-        CaseData caseDataCopy = details.getData().toBuilder().build();
+        CaseData data = details.getData();
+        CaseData caseDataCopy = data.toBuilder().build();
         OrderSummary orderSummary = paymentService.getOrderSummary();
         caseDataCopy.setApplicationFeeOrderSummary(orderSummary);
 
-        ListValue<Payment> paymentListValue = createPendingPayment(orderSummary.getPaymentTotal());
-        caseDataCopy.setPayments(singletonList(paymentListValue));
+        if (data.getPayments() == null || data.getPayments().isEmpty()) {
```

Comment:
I'm not sure about locking this behind this if statement. If there is a failed payment it won't execute. We might fail, go back and then want to add another payment

Reply:
This seems to be always replacing any existing payments with a mock payment, so I wasn't able to transition without checking to make sure it does replace an existing payments with a mock payment.

Reply:
I'm working on creating you a real one but I'm starting to second guess myself as to whether it should be done in the frontend